src/HieDb/Create.hs

```diff
@@ -352,25 +387,25 @@ addRefsFromLoaded_unsafe
         let sourceOnlyNodeInfo = SourcedNodeInfo $ M.delete originToDrop sniMap
         in Node sourceOnlyNodeInfo sp (map (dropNodeInfos originToDrop) children)
 
-  execute conn "INSERT INTO mods VALUES (?,?,?,?,?,?,?)" modrow
+  runStatementFor_ (insertModsStatement db) modrow
 
   let AstInfo refsSrc declsSrc importsSrc = genAstInfo path smod SourceInfo refmapSourceOnly
       AstInfo refsGen declsGen importsGen = genAstInfo path smod GeneratedInfo refmapGeneratedOnly
 
   unless (skipRefs skipOptions) $
-    executeMany conn "INSERT INTO refs VALUES (?,?,?,?,?,?,?,?,?)" (refsSrc <> refsGen)
+    mapM_ (runStatementFor_ (insertRefsStatement db)) (refsSrc <> refsGen)
   unless (skipDecls skipOptions) $
-    executeMany conn "INSERT INTO decls VALUES (?,?,?,?,?,?,?)" (declsSrc <> declsGen)
+    mapM_ (runStatementFor_ (insertDeclsStatement db)) (declsSrc <> declsGen)
   unless (skipImports skipOptions) $
-    executeMany conn "INSERT INTO imports VALUES (?,?,?,?,?,?)" (importsSrc <> importsGen)
+    mapM_ (runStatementFor_ (insertImportsStatement db)) (importsSrc <> importsGen)
 
   let defs = genDefRow path smod refmapAll
   unless (skipDefs skipOptions) $
-      executeMany conn "INSERT INTO defs VALUES (?,?,?,?,?,?)" defs
+    mapM_ (runStatementFor_ (insertDefsStatement db)) defs
 
   let exports = generateExports path $ hie_exports hf
   unless (skipExports skipOptions) $
-    executeMany conn "INSERT INTO exports VALUES (?,?,?,?,?,?,?,?)" exports
+    mapM_ (runStatementFor_ (insertExportsStatement db)) exports
```

Comment:
I am surprised this is faster than the batching version, perhaps a comment saying this was benchmarked would be nice :)